src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -276,7 +277,9 @@ class DefaultEntityProvider(
                 nextId = recycled.id + 1
             }
 
-            recycled
+            recycled.copy(version = recycled.version + 1)
+        }.also {
+            entities[it.id] = it
```

Comment:
That's interesting. I didn't know that you can do that in Kotlin. I learned something new, thanks! :)